File: source/models/supabase/panel.py
from datetime import datetime
from enum import Enum
from typing import ClassVar, Dict, Optional, List
from uuid import UUID
from pydantic import Field
from source.models.supabase.supabase_model import SupabaseModel


class ProcessState(str, Enum):
    none = "none"
    waiting = "waiting"
    processing = "processing"
    failed = "failed"
    done = "done"


class PanelAudio(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "panel_audio"
    id: Optional[UUID] = Field(default=None)
    panel_id: UUID
    transcript_id: UUID
    title: Optional[str] = Field(default=None)
    tags: Optional[List[str]] = Field(default=None)
    file: Optional[str] = Field(default=None)
    bucket_id: str = Field(default="public_panels")
    process_state: Optional[ProcessState] = Field(default=None)
    process_fail_message: Optional[str] = Field(default=None)
    metadata: Optional[Dict] = Field(default=None)
    is_public: Optional[bool] = Field(default=False)
    disabled: Optional[bool] = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)


class PanelDiscussion(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "panel_discussion"
    id: Optional[UUID] = Field(default=None)
    title: Optional[str] = Field(default=None)
    tags: Optional[List[str]] = Field(default=None)
    metadata: Optional[Dict] = Field(default=None)
    is_public: Optional[bool] = Field(default=False)
    disabled: Optional[bool] = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)


class PanelTranscript(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "panel_transcript"
    id: Optional[UUID] = Field(default=None)
    panel_id: UUID
    title: Optional[str] = Field(default=None)
    tags: Optional[List[str]] = Field(default=None)
    file: Optional[str] = Field(default=None)
    bucket_id: str = Field(default="public_panels")
    type: Optional[str] = Field(default=None)
    transcript: Optional[Dict] = Field(default=None)
    process_state: Optional[ProcessState] = Field(default=None)
    process_fail_message: Optional[str] = Field(default=None)
    generation_interval: Optional[int] = Field(default=None)
    generation_parent: Optional[UUID] = Field(default=None)
    metadata: Optional[Dict] = Field(default=None)
    is_public: Optional[bool] = Field(default=False)
    disabled: Optional[bool] = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)


class PanelTranscriptOrder(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "panel_transcript_order"
    id: Optional[UUID] = Field(default=None)
    panel_id: UUID
    transcript_id: Optional[UUID] = Field(default=None)
    before_id: Optional[UUID] = Field(default=None)
    after_id: Optional[UUID] = Field(default=None)
    data: Optional[Dict] = Field(default=None)
    is_public: Optional[bool] = Field(default=False)
    disabled: Optional[bool] = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    disabled: bool = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)


class PanelTranscriptSourceReference(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "panel_transcript_source_reference"
    id: Optional[UUID] = Field(default=None)
    transcript_id: UUID
    source_id: UUID
    type: Optional[str] = Field(default=None)
    data: Optional[Dict] = Field(default=None)
    is_public: Optional[bool] = Field(default=False)
    disabled: Optional[bool] = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    disabled: bool = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)

File: lib_js/models/supabase/panel.js
// Import necessary modules and base class
import { SupabaseModel, Enum } from "./supabaseModel.js";

// Define ProcessState Enum
class ProcessStateEnum extends Enum {
    constructor() {
        super();
        this.NONE = "none";
        this.WAITING = "waiting";
        this.PROCESSING = "processing";
        this.FAILED = "failed";
        this.DONE = "done";
        Object.freeze(this);
    }
}

export const ProcessState = new ProcessStateEnum();

// Define PanelAudioModel
export class PanelAudioModel extends SupabaseModel {
    static TABLE_NAME = "panel_audio";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: false, dbColumn: "id" },
        panelId: { type: "uuid", required: true, dbColumn: "panel_id" },
        transcriptId: {
            type: "uuid",
            required: true,
            dbColumn: "transcript_id"
        },
        title: { type: "string", required: false, dbColumn: "title" },
        tags: { type: "array", required: false, dbColumn: "tags" },
        file: { type: "string", required: false, dbColumn: "file" },
        bucketId: {
            type: "string",
            required: true,
            dbColumn: "bucket_id",
            default: "public_panels"
        },
        processState: {
            type: ProcessState,
            required: false,
            dbColumn: "process_state"
        },
        processFailMessage: {
            type: "string",
            required: false,
            dbColumn: "process_fail_message"
        },
        metadata: { type: "json", required: false, dbColumn: "metadata" },
        isPublic: {
            type: "boolean",
            required: false,
            dbColumn: "is_public",
            default: false
        },
        disabled: {
            type: "boolean",
            required: false,
            dbColumn: "disabled",
            default: false
        },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define PanelDiscussionModel
export class PanelDiscussionModel extends SupabaseModel {
    static TABLE_NAME = "panel_discussion";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: false, dbColumn: "id" },
        title: { type: "string", required: false, dbColumn: "title" },
        tags: { type: "array", required: false, dbColumn: "tags" },
        metadata: { type: "json", required: false, dbColumn: "metadata" },
        isPublic: {
            type: "boolean",
            required: false,
            dbColumn: "is_public",
            default: false
        },
        disabled: {
            type: "boolean",
            required: false,
            dbColumn: "disabled",
            default: false
        },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define PanelTranscriptModel
export class PanelTranscriptModel extends SupabaseModel {
    static TABLE_NAME = "panel_transcript";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: false, dbColumn: "id" },
        panelId: { type: "uuid", required: true, dbColumn: "panel_id" },
        title: { type: "string", required: false, dbColumn: "title" },
        tags: { type: "array", required: false, dbColumn: "tags" },
        file: { type: "string", required: false, dbColumn: "file" },
        bucketId: {
            type: "string",
            required: true,
            dbColumn: "bucket_id",
            default: "public_panels"
        },
        type: { type: "string", required: false, dbColumn: "type" },
        transcript: { type: "json", required: false, dbColumn: "transcript" },
        processState: {
            type: ProcessState,
            required: false,
            dbColumn: "process_state"
        },
        processFailMessage: {
            type: "string",
            required: false,
            dbColumn: "process_fail_message"
        },
        generationCronjob: {
            type: "string",
            required: false,
            dbColumn: "generation_cronjob"
        },
        generationParent: {
            type: "uuid",
            required: false,
            dbColumn: "generation_parent"
        },
        metadata: { type: "json", required: false, dbColumn: "metadata" },
        isPublic: {
            type: "boolean",
            required: false,
            dbColumn: "is_public",
            default: false
        },
        disabled: {
            type: "boolean",
            required: false,
            dbColumn: "disabled",
            default: false
        },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define PanelTranscriptOrderModel
export class PanelTranscriptOrderModel extends SupabaseModel {
    static TABLE_NAME = "panel_transcript_order";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: false, dbColumn: "id" },
        panelId: { type: "uuid", required: true, dbColumn: "panel_id" },
        transcriptId: {
            type: "uuid",
            required: false,
            dbColumn: "transcript_id"
        },
        beforeId: { type: "uuid", required: false, dbColumn: "before_id" },
        afterId: { type: "uuid", required: false, dbColumn: "after_id" },
        data: { type: "json", required: false, dbColumn: "data" },
        isPublic: {
            type: "boolean",
            required: false,
            dbColumn: "is_public",
            default: false
        },
        disabled: {
            type: "boolean",
            required: false,
            dbColumn: "disabled",
            default: false
        },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define PanelTranscriptSourceReferenceModel
export class PanelTranscriptSourceReferenceModel extends SupabaseModel {
    static TABLE_NAME = "panel_transcript_source_reference";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: false, dbColumn: "id" },
        transcriptId: {
            type: "uuid",
            required: true,
            dbColumn: "transcript_id"
        },
        sourceId: { type: "uuid", required: true, dbColumn: "source_id" },
        type: { type: "string", required: false, dbColumn: "type" },
        data: { type: "json", required: false, dbColumn: "data" },
        isPublic: {
            type: "boolean",
            required: false,
            dbColumn: "is_public",
            default: false
        },
        disabled: {
            type: "boolean",
            required: false,
            dbColumn: "disabled",
            default: false
        },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}
